src/main/java/net/jung/chess/Board.java

```diff
@@ -1,132 +1,142 @@
     }
 
-    public int whitePieceListSize() {
-        return whitePieceList.size();
+    public void initializeEmptyBoard() {
+        for (int i = 0; i < RANKS_ON_BOARD; i++) {
+            rankList.add(Rank.initializeBlankRank());
+        }
     }
 
-    public int blackPieceListSize() {
-        return blackPieceList.size();
+    public String boardLayoutToString() {
+        StringBuilder boardLayout = new StringBuilder();
+        for (Rank rank : rankList) {
+            boardLayout.append(appendNewLine(rank.rankLayoutToString()));
+        }
+        return boardLayout.toString();
     }
 
-    public Piece findWhitePiece(int index) {
-        return whitePieceList.get(index);
+    public void reset() {
+        rankList.clear();
     }
 
-    public Piece findBlackPiece(int index) {
-        return blackPieceList.get(index);
+    public Rank getRank(int rankNum) {
+        return rankList.get(RANKS_ON_BOARD - rankNum);
     }
 
-    public void initialize() {
-        initializeBlackPieces();
-        initializePawns(Color.BLACK);
+    public Piece findPiece(Position position) {
+        return getRank(position.getRankIndex()).getPiece(position.getFileIndex());
+    }
 
-        initializePawns(Color.WHITE);
-        initializeWhitePieces();
+    public void addNewPiece(Position position, Piece newPiece) {
+        getRank(position.getRankIndex()).replacePiece(position.getFileIndex(), newPiece);
     }
 
-    public void initializePawns(Color color) {
-        int maxPawnsSize = 8;
-        if(color == Color.WHITE) {
-            for (int i = 0; i < maxPawnsSize; i++) {
-            addWhitePiece(Piece.createWhitePawn());
-            }
-        return;
-        }
+    public void movePiece(Position startPosition, Position endPosition) {
+        Piece foundPiece = findPiece(startPosition);
 
-        for(int j = 0; j < maxPawnsSize; j++) {
-            addBlackPiece(Piece.createBlackPawn());
+        if (foundPiece.isPiece()) {
+            getRank(startPosition.getRankIndex()).replacePiece(startPosition.getFileIndex(), Piece.createBlank());
+            getRank(endPosition.getRankIndex()).replacePiece(endPosition.getFileIndex(), foundPiece);
         }
     }
 
-    public void initializeBlackPieces(){
-        addBlackPiece(Piece.createBlackRook());
-        addBlackPiece(Piece.createBlackKnight());
-        addBlackPiece(Piece.createBlackBishop());
-        addBlackPiece(Piece.createBlackQueen());
-        addBlackPiece(Piece.createBlackKing());
-        addBlackPiece(Piece.createBlackBishop());
-        addBlackPiece(Piece.createBlackKnight());
-        addBlackPiece(Piece.createBlackRook());
-    }
+    public double calculatePoints(Piece.Color color) {
+        double points = 0;
+
+        List<Piece> colorPieceList = getPieceListByColor(color);
+        for (int i = 0; i < colorPieceList.size(); i++) {
+            points += colorPieceList.get(i).getDefaultPoint();
+        }
```

Comment:
```suggestion
        points = colorPieceList.stream()
                .mapToDouble(Piece::getDefaultPoint)
                .sum();
```
여기도 stream으로 바꾸거나 enhanced for loop를 사용할 수 있겠네요

Reply:
여기서도 마찬가지로 일단 수정한 후에 스트림을 제대로 공부하겠습니다!